src/android/com/microsoftopentechnologies/intellij/forms/TableEntityForm.java

```diff
@@ -0,0 +1,410 @@
+package com.microsoftopentechnologies.intellij.forms;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.ComboBoxTableRenderer;
+import com.intellij.ui.JBColor;
+import com.intellij.util.ui.table.ComboBoxTableCellEditor;
+import com.microsoftopentechnologies.intellij.helpers.UIHelper;
+import com.microsoftopentechnologies.intellij.helpers.azure.AzureCmdException;
+import com.microsoftopentechnologies.intellij.helpers.azure.sdk.AzureSDKManagerImpl;
+import com.microsoftopentechnologies.intellij.helpers.storage.TableFileEditor;
+import com.microsoftopentechnologies.intellij.model.storage.StorageAccount;
+import com.microsoftopentechnologies.intellij.model.storage.TableEntity;
+import org.jdesktop.swingx.JXMonthView;
+import org.jdesktop.swingx.calendar.DateSelectionModel;
+import org.jetbrains.annotations.NotNull;
+import sun.reflect.misc.ReflectUtil;
+import sun.swing.SwingUtilities2;
+
+import javax.swing.*;
+import javax.swing.border.LineBorder;
+import javax.swing.table.*;
+import java.awt.*;
+import java.awt.event.*;
+import java.lang.reflect.Constructor;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class TableEntityForm extends JDialog {
+    private JPanel contentPane;
+    private JButton buttonOK;
+    private JButton buttonCancel;
+    private JButton addPropertyButton;
+    private JTable propertiesTable;
+
+    private Project project;
+    private TableEntity tableEntity;
+    private StorageAccount storageAccount;
```

Comment:
TableEntity is a new type.

StorageAccount is not a new type, it was created when we developed the VM creation. What we did was move it to the storage namespace, since it made more sense for it to be there.